mne/io/egi/egimff.py

```diff
@@ -0,0 +1,368 @@
+        Headers.
+    mff_events : dictionary.
+        Events.
+    """
+    mff_hdr = read_mff_header(input_fname)
+    with open(input_fname + '/signal1.bin', 'rb') as fid:
+        version = np.fromfile(fid, np.int32, 1)[0]
+
+    # if version > 6 & ~np.bitwise_and(version, 6):
+    #    version = version.byteswap().astype(np.uint32)
+    # else:
+    #    ValueError('Watchout. This does not seem to be a simple '
+    #               'binary EGI file.')
+
+    time_n = dateutil.parser.parse(mff_hdr['date'])
+    info = dict(   # leer del info.xml desde read_mff_header
+        version=version,  # duda
+        year=int(time_n.strftime('%Y')),           # duda
+        month=int(time_n.strftime('%m')),          # duda
+        day=int(time_n.strftime('%d')),
+        hour=int(time_n.strftime('%H')),
+        minute=int(time_n.strftime('%M')),
+        second=int(time_n.strftime('%S')),
+        millisecond=int(time_n.strftime('%f')),
+        samp_rate=mff_hdr['Fs'],
+        n_channels=mff_hdr['nChans'],
+        gain=0,           # duda
+        bits=0,           # duda
+        value_range=0)    # duda
+    unsegmented = 1 if mff_hdr['nTrials'] == 1 else 0
+    precision = 4
+    # np.bitwise_and(version,6)
+    if precision == 0:
+        RuntimeError('Floating point precision is undefined.')
+    if unsegmented:
+        info.update(dict(n_categories=0,
+                         n_segments=1,
+                         n_samples=mff_hdr['nSamples'],
+                         n_events=0,
+                         event_codes=[],
+                         category_names=[],
+                         category_lengths=[],
+                         pre_baseline=0))
+    else:
+        raise NotImplementedError('Only continuos files are supported')
+    info['unsegmented'] = unsegmented
+    info['dtype'], info['orig_format'] = {2: ('>i2', 'short'),
+                                          4: ('>f4', 'float'),
+                                          6: ('>f8', 'double')}[precision]
+    info['dtype'] = np.dtype(info['dtype'])
+    return info, mff_hdr
+
+
+def _read_events(input_fname, hdr, info):
+    """Read events for the record.
+
+    Parameters
+    ----------
+    input_fname : str
+        The file path.
+    hdr : dict
+        Dictionary with the headers got from read_mff_header.
+    info : dict
+        Header info array.
+    """
+    mff_events, event_codes = read_mff_events(input_fname, hdr)
+    info['n_events'] = len(event_codes)
+    info['event_codes'] = np.asarray(event_codes).astype('<U4')
+    events = np.zeros([info['n_events'],
+                      info['n_segments'] * info['n_samples']])
+    for n, event in enumerate(event_codes):
+        for i in mff_events[event]:
+            if i > events.shape[1]:
+                warn('Event outside data range (%ss).' % (i /
+                                                          info['samp_rate']))
+                continue
+            events[n][i] = 2**n
```

Comment:
Okay then we should probably have a parameter for this. `event_numbering='linear' | 'log'` or `'consecutive' | 'power'` or something. If we want to make `2**n` the default here, it should probably be part of this PR, or as part of an immediate follow-up.